exchange/adapter_util.go

```diff
@@ -0,0 +1,150 @@
+	"github.com/prebid/prebid-server/adapters/ix"
+	"github.com/prebid/prebid-server/adapters/lifestreet"
+	"github.com/prebid/prebid-server/adapters/pulsepoint"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+func BuildAdapters(client *http.Client, cfg *config.Configuration, infos adapters.BidderInfos, me pbsmetrics.MetricsEngine) (map[openrtb_ext.BidderName]adaptedBidder, []error) {
+	exchangeBidders, errs := buildExchangeBidders(cfg, infos, client, me)
+	if len(errs) > 0 {
+		return nil, errs
+	}
+
+	exchangeBiddersLegacy := buildExchangeBiddersLegacy(cfg.Adapters, infos)
+	for bidderName, bidder := range exchangeBiddersLegacy {
+		exchangeBidders[bidderName] = bidder
+	}
+
+	wrapWithMiddleware(exchangeBidders)
+
+	return exchangeBidders, nil
+}
+
+func buildExchangeBidders(cfg *config.Configuration, infos adapters.BidderInfos, client *http.Client, me pbsmetrics.MetricsEngine) (map[openrtb_ext.BidderName]adaptedBidder, []error) {
+	bidders, errs := buildBidders(cfg.Adapters, infos, newAdapterBuilders())
+	if len(errs) > 0 {
+		return nil, errs
+	}
+
+	exchangeBidders := make(map[openrtb_ext.BidderName]adaptedBidder, len(bidders))
+	for bidderName, bidder := range bidders {
+		exchangeBidders[bidderName] = adaptBidder(bidder, client, cfg, me, bidderName)
+	}
+
+	return exchangeBidders, nil
+
+}
+
+func buildBidders(adapterConfig map[string]config.Adapter, infos adapters.BidderInfos, builders map[openrtb_ext.BidderName]adapters.Builder) (map[openrtb_ext.BidderName]adapters.Bidder, []error) {
+	bidders := make(map[openrtb_ext.BidderName]adapters.Bidder)
+	var errs []error
+
+	for bidder, cfg := range adapterConfig {
+		bidderName, bidderNameFound := openrtb_ext.NormalizeBidderName(bidder)
+		if !bidderNameFound {
+			errs = append(errs, fmt.Errorf("%v: unknown bidder", bidder))
+			continue
+		}
+
+		// Ignore Legacy Bidders
+		if bidderName == openrtb_ext.BidderIx || bidderName == openrtb_ext.BidderLifestreet || bidderName == openrtb_ext.BidderPulsepoint {
+			continue
+		}
+
+		info, infoFound := infos[string(bidderName)]
+		if !infoFound {
+			errs = append(errs, fmt.Errorf("%v: bidder info not found", bidder))
+			continue
+		}
+
+		builder, builderFound := builders[bidderName]
+		if !builderFound {
+			errs = append(errs, fmt.Errorf("%v: builder not registered", bidder))
+			continue
+		}
+
+		if info.Status == adapters.StatusActive {
+			bidderInstance, builderErr := builder(bidderName, cfg)
+			if builderErr != nil {
+				errs = append(errs, fmt.Errorf("%v: %v", bidder, builderErr))
+				continue
+			}
+
+			bidderWithInfoEnforcement := adapters.EnforceBidderInfo(bidderInstance, info)
+
+			bidders[bidderName] = bidderWithInfoEnforcement
+		}
+	}
+
+	return bidders, errs
+}
+
+func buildExchangeBiddersLegacy(adapterConfig map[string]config.Adapter, infos adapters.BidderInfos) map[openrtb_ext.BidderName]adaptedBidder {
+	bidders := make(map[openrtb_ext.BidderName]adaptedBidder, 3)
```

Comment:
We're actively working now to add OpenRTB adapters for these 3. Once that happens, this method can be removed and this file's code can be simplified.